pkg/controller/hibernation/ibmcloud_actuator_test.go

```diff
@@ -0,0 +1,231 @@
+					func(instances []vpcv1.Instance) {
+						for _, i := range instances {
+							assert.True(t, *i.Status == "stopped" || *i.Status == "stopping" || *i.Status == "deleting")
+						}
+					},
+				).Return(nil)
+			},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+			ibmCloudClient := mockibmclient.NewMockAPI(ctrl)
+			setupIBMCloudClientInstances(ibmCloudClient, test.instances)
+			if test.setupClient != nil {
+				test.setupClient(t, ibmCloudClient)
+			}
+			actuator := testIBMCloudActuator(ibmCloudClient)
+			var err error
+			switch test.testFunc {
+			case "StopMachines":
+				err = actuator.StopMachines(testClusterDeployment(), nil, log.New())
+			case "StartMachines":
+				err = actuator.StartMachines(testClusterDeployment(), nil, log.New())
+			default:
+				t.Fatal("Invalid function to test")
+			}
+			assert.Nil(t, err)
+			ctrl.Finish()
+		})
+	}
+
+}
+
+func TestIBMCloudMachinesStoppedAndRunning(t *testing.T) {
+	tests := []struct {
+		name        string
+		testFunc    string
+		expected    bool
+		instances   map[string]int
+		setupClient func(*testing.T, *mockibmclient.MockAPI)
+	}{
+		{
+			name:      "Stopped - All machines stopped or terminated",
+			testFunc:  "MachinesStopped",
+			expected:  true,
+			instances: map[string]int{"deleting": 3, "stopped": 2},
+		},
+		{
+			name:      "Stopped - Some machines pending",
+			testFunc:  "MachinesStopped",
+			expected:  false,
+			instances: map[string]int{"deleting": 3, "stopped": 2, "pending": 2},
+		},
+		{
+			name:      "Stopped - machines running",
+			testFunc:  "MachinesStopped",
+			expected:  false,
+			instances: map[string]int{"running": 3, "deleting": 2},
+		},
+		{
+			name:      "Running - All machines running",
+			testFunc:  "MachinesRunning",
+			expected:  true,
+			instances: map[string]int{"running": 3},
+		},
+		{
+			name:      "Running - Some machines pending",
+			testFunc:  "MachinesRunning",
+			expected:  false,
+			instances: map[string]int{"running": 3, "pending": 1},
+		},
+		{
+			name:      "Running - Some machines stopped or shutting-down",
+			testFunc:  "MachinesRunning",
+			expected:  false,
+			instances: map[string]int{"running": 3, "deleting": 2, "stopped": 1, "stopping": 3},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+			ibmCloudClient := mockibmclient.NewMockAPI(ctrl)
+			setupIBMCloudClientInstances(ibmCloudClient, test.instances)
+			if test.setupClient != nil {
+				test.setupClient(t, ibmCloudClient)
+			}
+			actuator := testIBMCloudActuator(ibmCloudClient)
+			var err error
+			var result bool
+			switch test.testFunc {
+			case "MachinesStopped":
+				result, _, err = actuator.MachinesStopped(testClusterDeployment(), nil, log.New())
+			case "MachinesRunning":
+				result, _, err = actuator.MachinesRunning(testClusterDeployment(), nil, log.New())
```

Comment:
It would be cool if these could also validate the "list of machines" return. The validator would have to sort the actual & expected (I can't find an `assert.EqualElements` or similar).